### PR TITLE
refactor(meta): commit finish streaming job in barrier manager

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/controller/streaming_job.rs [hunk 1]
```diff
@@ -473,6 +473,15 @@ impl CatalogController {
         if let Some(source_id) = associated_source_id {
             Object::delete_by_id(source_id).exec(&txn).await?;
         }
+
+        for tx in inner.table_id_to_tx.remove(&job_id).into_iter().flatten() {
```
Comment (kwannoel) [inline src/meta/src/controller/streaming_job.rs:477]: This seems to suggest there's more than one tx per table id. Should it be `if let Some(tx) = ...` instead?

**Review by kwannoel — APPROVED** [review]
Changes LGTM.
Instead of the catalog manager having to maintain a version map, we just check if the table catalog is in the catalog controller / catalog manager. This is a much better improvement.

There's two cases to consider:
1. The streaming job is finished before the manager inserts the notification channel. In this case the catalog must be present.
2. The streaming job is not finished. In this case we can just insert the notification channel.

The case where streaming job finishes _after_ checking if catalog is present, and _before_ catalog is committed cannot happen, because we already have a write guard inside our `finish` methods, for the catalog writer, so it prevents this sequence of events.

This PR also contains some improvements to the recovery functions for v1 and v2, collapsing the parameters into a single map. Also LGTM.

Thanks for your work!

**Comment by kwannoel** [comment]
Changed back to `merge into main`. Changed the branch target for easier reviewing.

**Comment by wenym1** [comment]
> Changed back to `merge into main`. Changed the branch target for easier reviewing previously.

I will merge this PR to the original branch for review from other peers.

**Pull request merged**